crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -103,13 +124,43 @@ async fn configure_dsn(
     enable_dsn: bool,
     listen_on: Vec<Multiaddr>,
     bootstrap_nodes: Vec<Multiaddr>,
-) -> Result<(Option<Node>, Option<NodeRunner>), anyhow::Error> {
+    record_cache_size: usize,
+    record_cache_db_path: Option<String>,
+) -> Result<(Option<Node>, Option<NodeRunner<ConfiguredRecordStore>>), anyhow::Error> {
     if !enable_dsn {
         info!("No DSN configured.");
         return Ok((None, None));
     }
 
-    let config = Config {
+    let record_cache_size = NonZeroUsize::new(record_cache_size).unwrap_or(
+        NonZeroUsize::new(MAX_KADEMLIA_RECORDS_NUMBER)
+            .expect("We don't expect an error on manually set value."),
+    );
+
+    let record_cache_db_path = record_cache_db_path
+        .and_then(|path| {
+            let path_result = PathBuf::from_str(&path);
+
+            if let Err(ref err) = path_result {
+                error!(
+                    ?err,
+                    "Invalid record cache DB path. Temp directory used instead."
+                );
```

Comment:
This is a fatal error. It is extremely surprising if application just moves on with temporary directory instead (that is often mounted as tmpfs and can cause system to run out of memory). Though if you change the type in function signature you wouldn't need to bother with this.